packages/lib-classifier/src/store/SubjectStore/SingleTextSubject/SingleTextSubject.js

```diff
@@ -1,11 +1,65 @@
-import { types } from 'mobx-state-tree'
-import createLocationCounts from '@helpers/createLocationCounts'
+import { flow, types } from 'mobx-state-tree'
+import request from 'superagent'
+import asyncStates from '@zooniverse/async-states'
+
+import { createLocationCounts } from '@helpers'
 import Subject from '../Subject'
 
-const SingleTextSubject = types
-  .refinement('SingleTextSubject', Subject, subject => {
-    const counts = createLocationCounts(subject)
-    return subject.locations.length === 1 && counts.text === 1
+const TextSubject = types
+  .model('TextSubject', {
+    content: types.maybeNull(types.string),
+    contentLoadingState: types.optional(types.enumeration('contentLoadingState', asyncStates.values), asyncStates.initialized),
+    error: types.maybeNull(types.frozen({}))
   })
+  .actions(self => {
+    function afterAttach () {
+      if (self.contentLoadingState === asyncStates.initialized) {
+        self.fetchContent()
+      }
+    }
+
+    function getSubjectUrl () {
+      // Find locations that have a text/plain MIME type.
+      const textLocation = self.locations.find(l => l['text/plain']) || {}
```

Comment:
This is true for PH-TESS subjects, so we should make sure that they don't accidentally load using this model.

Reply:
the workflow configuration determines the subject viewer if it's defined, per https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/store/SubjectStore/Subject/Subject.js#L54.

https://local.zooniverse.org:8080/?project=7929&env=production&demo=true I think is working as expected.